marimo/_output/formatters/arviz_formatters.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+from typing import Any, Union
+from marimo._messaging.mimetypes import KnownMimeType
+from marimo._output.formatters.formatter_factory import FormatterFactory
+import matplotlib.pyplot as plt
+import numpy as np
```

Comment:
this can't be at the top-level import since its optional. instead it should be in a typechecking block:
```
if TYPE_CHECKING:
    import numpy as np
```

and then imported again, whenever its used at runtime.

Reply:
import `typing` and put the import statements under `TYPE_CHECKING`; although running `ruff check` hints at not doing this.

Reply:
Why does ruff complain? Are you sure it breaks ci? This is required before merging

Reply:
> Why does ruff complain? Are you sure it breaks ci? This is required before merging

I'm not sure why; running it locally seems to cause issues with comments on certain lines being too long: `E501 Line too long` and `= help: Move out of type-checking block` for the 2 import statements inside `TYPE_CHECKING`

Reply:
> Why does ruff complain? Are you sure it breaks ci? This is required before merging

Resolved most ruff errors now except the import statements under the `TYPE_CHECKING` part.